### ISSUE TITLE
Unable to run SMBD test cases for version 3.20.1.0

### ISSUE THREAD

**Issue by vinitagnihotri** [microsoft/WindowsProtocolTestSuites#230]
Hi,
We are running our linux based SMB server in centos box. We have used test suite version 3.18.6 earlier and it worked for us. We have recently installed version 3.20.1.0, but it do not seem to connect to our server anymore. 

We can access shares over SMBD without any issue, but somehow test suite is not able to communicate. 

We could see following text for any test case

```
2020-04-14 20:02:57.511 [TestInProgress] Microsoft.Protocol.TestSuites.Smbd.TestSuite.SmbdNegotiate.BVT_SmbdNegotiate_Basic

2020-04-14 20:02:57.542 [Debug] Power Shell adapter: ISutProtocolControlAdapter, method: Initialize not found, skipped.

2020-04-14 20:02:57.557 [Comment] [TestPurpose]Verify SMBD negotiate request is working as expected using accepted values.

2020-04-14 20:02:57.557 [TestStep] Connect to server over RDMA

2020-04-14 20:02:57.651 [CheckFailed] Assert.AreEqual failed. Expected: <STATUS_SUCCESS (Microsoft.Protocols.TestTools.StackSdk.NtStatus)>, Actual: <STATUS_INVALID_PARAMETER (Microsoft.Protocols.TestTools.StackSdk.NtStatus)>. Status of SMBD connection is STATUS_INVALID_PARAMETER

2020-04-14 20:02:57.651 [Comment] at Microsoft.Protocol.TestSuites.Smbd.TestSuite.SmbdNegotiate.BasicNegotiate(UInt16 creditsRequested, UInt32 preferredSendSize, UInt32 maxReceiveSize, UInt32 maxFragmentedSize, SmbdNegotiateResponse& response) in C:\agent\_work\6\b\TestSuites\MS-SMBD\src\TestSuite\SMBD\SmbdNegotiate.cs:line 535 at Microsoft.Protocol.TestSuites.Smbd.TestSuite.SmbdNegotiate.BVT_SmbdNegotiate_Basic() in C:\agent\_work\6\b\TestSuites\MS-SMBD\src\TestSuite\SMBD\SmbdNegotiate.cs:line 52

2020-04-14 20:02:57.682 [TestFailed] Microsoft.Protocol.TestSuites.Smbd.TestSuite.SmbdNegotiate.BVT_SmbdNegotiate_Basic
```

From our server end we could see connection rejected by rdma subsystem. But older version of MSFT 3.18.6 seems to work ok.

Can you please help in this regard?

**Comment by huiren-jiang** [comment]
Hi vinitagnihotri,

Thanks for reporting this issue.

The related RDMA layer code has not been updated since 3.19.6.0 and the attached log does not contain enough information. Could you try to set `RdmaLayerLoggingEnabled` to `true` in `MS-SMBD_ServerTestSuite.ptfconfig`? This step will generate more logs to help us to figure out the reason for the connection failure.

Best Regards,

Windows Protocol Test Suites Development Team

**Comment by vinitagnihotri** [comment]
For some reason I was not able to attach it as file.
therefore I am pasting contents here,.

BVT_SmbdNegotiate_Basic
Start Time: 2020-04-17 04:37:30.635
End Time: 2020-04-17 04:37:32.182
Result: Failed
 ErrorStackTrace 
at Microsoft.Protocols.TestTools.AsynchronousErrorProcessor.Process() 
at Microsoft.Protocols.TestTools.AsynchronousErrorProcessor.ReportAsyncException(Exception checkerException) 
at Microsoft.Protocols.TestTools.Checking.DefaultChecker`2.AreEqual[T](T expected, T actual, String message, Object[] parameters) 
at Microsoft.Protocol.TestSuites.Smbd.TestSuite.SmbdNegotiate.BasicNegotiate(UInt16 creditsRequested, UInt32 preferredSendSize, UInt32 maxReceiveSize, UInt32 maxFragmentedSize, SmbdNegotiateResponse& response) in C:\agent\_work\6\b\TestSuites\MS-SMBD\src\TestSuite\SMBD\SmbdNegotiate.cs:line 535 
at Microsoft.Protocol.TestSuites.Smbd.TestSuite.SmbdNegotiate.BVT_SmbdNegotiate_Basic() in C:\agent\_work\6\b\TestSuites\MS-SMBD\src\TestSuite\SMBD\SmbdNegotiate.cs:line 52
 ErrorMessage 
Assert.AreEqual failed. Expected: <STATUS_SUCCESS (Microsoft.Protocols.TestTools.StackSdk.NtStatus)>, Actual: <STATUS_INVALID_PARAMETER (Microsoft.Protocols.TestTools.StackSdk.NtStatus)>. Status of SMBD connection is STATUS_INVALID_PARAMETER 
===== Stack Trace ===== 
at Microsoft.Protocol.TestSuites.Smbd.TestSuite.SmbdNegotiate.BVT_SmbdNegotiate_Basic() in C:\agent\_work\6\b\TestSuites\MS-SMBD\src\TestSuite\SMBD\SmbdNegotiate.cs:line 52 
=== End of Stack Trace ===
 StandardOut 
Log Filter  
2020-04-17 10:07:31.682 [TestInProgress] Microsoft.Protocol.TestSuites.Smbd.TestSuite.SmbdNegotiate.BVT_SmbdNegotiate_Basic 
2020-04-17 10:07:31.713 [Debug] Power Shell adapter: ISutProtocolControlAdapter, method: Initialize not found, skipped. 
2020-04-17 10:07:31.729 [Comment] [TestPurpose]Verify SMBD negotiate request is working as expected using accepted values. 
2020-04-17 10:07:31.745 [TestStep] Connect to server over RDMA 
2020-04-17 10:07:31.829 [Debug] Loading the providers of registered network drivers. 
2020-04-17 10:07:31.885 [Debug] 12 providers of registered network drivers have been load, 
2020-04-17 10:07:31.885 [Debug] Load provider 1: E:\Windows\system32\mswsock.dll 
2020-04-17 10:07:31.900 [Debug] Load provider 2: E:\Windows\system32\mswsock.dll 
2020-04-17 10:07:31.900 [Debug] Load provider 3: E:\Windows\system32\mswsock.dll 
2020-04-17 10:07:31.900 [Debug] Load provider 4: E:\Windows\system32\mswsock.dll 
2020-04-17 10:07:31.900 [Debug] Load provider 5: E:\Windows\system32\mswsock.dll 
2020-04-17 10:07:31.900 [Debug] Load provider 6: E:\Windows\system32\mswsock.dll 
2020-04-17 10:07:31.900 [Debug] Load provider 7: E:\Windows\system32\mswsock.dll 
2020-04-17 10:07:31.900 [Debug] Load provider 8: E:\Windows\system32\mswsock.dll 
2020-04-17 10:07:31.900 [Debug] Load provider 9: E:\Windows\system32\mswsock.dll 
2020-04-17 10:07:31.900 [Debug] Load provider 10: E:\Windows\system32\mswsock.dll 
2020-04-17 10:07:31.900 [Debug] Load provider 11: E:\Windows\system32\mlx4nd.dll 
2020-04-17 10:07:31.900 [Debug] Load provider 12: E:\Windows\system32\mlx4nd.dll 
2020-04-17 10:07:31.900 [Debug] Loading providers is completed 
2020-04-17 10:07:31.900 [Debug] Try to open adapter from provider "E:\Windows\system32\mlx4nd.dll" with specific IP Address: "172.16.1.3" 
2020-04-17 10:07:31.916 [Debug] Total 1 addresses supported by the provider "E:\Windows\system32\mlx4nd.dll": 
2020-04-17 10:07:31.916 [Debug] 1. InterNetwork: 172.16.1.3 
2020-04-17 10:07:31.916 [Debug] Adapter on IP address "172.16.1.3" is open via provider 'E:\Windows\system32\mlx4nd.dll'. 
2020-04-17 10:07:31.963 [CheckFailed] Assert.AreEqual failed. Expected: <STATUS_SUCCESS (Microsoft.Protocols.TestTools.StackSdk.NtStatus)>, Actual: <STATUS_INVALID_PARAMETER (Microsoft.Protocols.TestTools.StackSdk.NtStatus)>. Status of SMBD connection is STATUS_INVALID_PARAMETER 
2020-04-17 10:07:32.025 [Comment] at Microsoft.Protocol.TestSuites.Smbd.TestSuite.SmbdNegotiate.BasicNegotiate(UInt16 creditsRequested, UInt32 preferredSendSize, UInt32 maxReceiveSize, UInt32 maxFragmentedSize, SmbdNegotiateResponse& response) in C:\agent\_work\6\b\TestSuites\MS-SMBD\src\TestSuite\SMBD\SmbdNegotiate.cs:line 535 at Microsoft.Protocol.TestSuites.Smbd.TestSuite.SmbdNegotiate.BVT_SmbdNegotiate_Basic() in C:\agent\_work\6\b\TestSuites\MS-SMBD\src\TestSuite\SMBD\SmbdNegotiate.cs:line 52 
2020-04-17 10:07:32.072 [TestFailed] Microsoft.Protocol.TestSuites.Smbd.TestSuite.SmbdNegotiate.BVT_SmbdNegotiate_Basic

**Comment by huiren-jiang** [comment]
Hi vinitagnihotri,

According to the log, some RDMA operations after opening adapter failed. This should be caused by mis-configuration of test suites, please try below steps:

1. Some RDMA related configuration in `MS-SMBD_ServerTestSuite.ptfconfig` might be not applied after you installed the latest version. For more information, you could refer to the [RDMA Capability Settings](https://github.com/microsoft/WindowsProtocolTestSuites/blob/staging/TestSuites/MS-SMBD/docs/MS-SMBD_ServerUserGuide.md#configuring-the-test-suite) part of the MS-SMBD user guide.

1. Check the detailed RDMA log on your server to identify the reason of rejection, if available.

1. Uninstall the SMBD test suites, and install the [MS-SMBD-TestSuite-ServerEP.zip](https://github.com/microsoft/WindowsProtocolTestSuites/files/4507148/MS-SMBD-TestSuite-ServerEP.zip)
 which we added more logs in commit 378ec1d. Set `RdmaLayerLoggingEnabled` to `true` and attach the new logs after re-run. 

Best Regards,

Windows Protocol Test Suites Development Team

**Comment by vinitagnihotri** [comment]
Hi,
I installed msi from zip file, but I am not able to run suite anymore. Please find image attached with mail.
![ms_suite_error](https://user-images.githubusercontent.com/54569394/80064476-c7431c80-8555-11ea-9362-7f368de9c013.png)

**Comment by huiren-jiang** [comment]
Hi vinitagnihotri,

According to your screenshot, the error in PTM is caused by the Protocol Test Framework installed is not the [latest](https://github.com/Microsoft/ProtocolTestFramework/releases/tag/1.0.7500.0). Also, we recommend you update to use latest PTM.

Please refer to the [Prerequisites](https://github.com/microsoft/WindowsProtocolTestSuites#prerequisites). Please make sure all old versions of test suites, Protocol Test Framework and Protocol Test Manager are uninstalled before you install the latest. 

Best Regards,

Windows Protocol Test Suites Development Team

**Comment by vinitagnihotri** [comment]
Thank you. I am facing difficulty installing appropriate test suite requirements. Let me try again and should update here.

**Comment by vinitagnihotri** [comment]
Hi,
I have logs that you have requested in a zip file.
I have also sending in message analyser capture file, this capture is of communication between MS-SMBClient over SMBDirect with our server. 

While we can access shares over SMBDirect from explorer, we could not run MS-SMBD for the same.


[sambdirect_Connection.zip](https://github.com/microsoft/WindowsProtocolTestSuites/files/4657397/sambdirect_Connection.zip)

**Comment by huiren-jiang** [comment]
Hi vinitagnihotri,

Observing the exception and stack trace in the attached `capture_from_Test_case.rtf`, it seems that ` InboundReadLimit` in `MS-SMBD_ServerTestSuite.ptfconfig` is not in a valid format of integer. Please modify it to some value according to [RDMA Capability Settings](https://github.com/microsoft/WindowsProtocolTestSuites/blob/staging/TestSuites/MS-SMBD/docs/MS-SMBD_ServerUserGuide.md#configuring-the-test-suite). After that, please rerun the test case to see if this issue could be resolved.

Best Regards,

Windows Protocol Test Suites Development Team

**Comment by vinitagnihotri** [comment]
Hi,
I tried with default values and with specific values yet results are same.
I tried values (16,16,1) and (63,63,10) (InboundEntries, OutboundEntries, InboundReadLimits respectively), result is same.

Here are values from mlx card capability, i am not sure how values you mention maps from Windows side, but I'll put them in as simple as I can from our perspective. May be you can help us map them to InboundEntries, OutboundEntries, InboundReadLimit.

**These are our values and its description and values in ( ). These values are queried from device.**
initiatorDepth - The maximum number of outstanding RDMA read that the local side will have to the remote side (128)

responderResources - The maximum number of outstanding RDMA read that the local side will accept from the remote side. (16)

max_qp_rd_atom - Maximum number of RDMA Reads & atomic operations that can be outstanding per QP with this device as the target, if such are supported (16)

max_res_rd_atom - Total maximum number of resources used for RDMA Reads & atomic operations by this device as the target, if such are supported (6814592)

max_qp_init_rd_atom - Maximum depth per QP for initiation of RDMA Read & atomic operations by this device, if such are supported (128) 

max_qp_wr - Maximum number of outstanding work requests on any Send or Receive Queue supported by this device (16351)

**Comment by huiren-jiang** [comment]
Hi vinitagnihotri,

Did the test cases failed the same because of `FormatException`? If yes, please check carefully if there is any unexpected characters within quotes. If not, could you attach the full log by setting `RdmaLayerLoggingEnabled` to `true` in `MS-SMBD_ServerTestSuite.ptfconfig`?

As I tested, test cases could pass even If I set all these three values to 1 since the values are smaller than supported maximum of RDMA adapter.

An auto-detection feature will be added to PTM, which could detect these capabilities of RDMA adapter. You could uninstall the old one and install [ProtocolTestManager_preview.zip](https://github.com/microsoft/WindowsProtocolTestSuites/files/4699867/ProtocolTestManager_preview.zip). After finishing auto-detection, these values will be configured automatically. Please refer to the screenshot attached below. According to the detected result against Mellanox adapter, these three values are `16351, 16351, 16`.
![ptm_screenshot](https://user-images.githubusercontent.com/27806452/83225102-92a82d80-a1b1-11ea-8334-d454f69f2b90.png)


Best Regards,

Windows Protocol Test Suites Development Team

**Comment by vinitagnihotri** [comment]
We took this zip file and ran auto detection, still MSFT reports RDMA not capable. Here are results,

[6/3/2020 1:10:08 PM] Ping Target SUT...
[6/3/2020 1:10:08 PM] Passed
[6/3/2020 1:10:08 PM] Check the OS version...
[6/3/2020 1:10:08 PM] [10.1.1.79] Connecting to remote server 10.1.1.79 failed with the following error message : The WinRM client cannot process the request. If the authentication scheme is different from Kerberos, or if the client computer is not joined to a domain, then HTTPS transport must be used or the destination machine must be added to the TrustedHosts configuration setting. Use winrm.cmd to configure TrustedHosts. Note that computers in the TrustedHosts list might not be authenticated. You can get more information about that by running the following command: winrm help config. For more information, see the about_Remote_Troubleshooting Help topic.
[6/3/2020 1:10:08 PM] Set platform to NonWindows.
[6/3/2020 1:10:08 PM] The format of return value is invalid!
[6/3/2020 1:10:08 PM] Failed
[6/3/2020 1:10:08 PM] Check the local adapters...
[6/3/2020 1:10:10 PM] Choose 10.1.1.80 as non-RDMA IP address of driver computer.
[6/3/2020 1:10:10 PM] Choose 172.16.1.3 as RDMA IP address of driver computer.
[6/3/2020 1:10:10 PM] Finished
[6/3/2020 1:10:10 PM] Check the supported SMB dialects of SUT...
[6/3/2020 1:10:10 PM] TryNegotiateDialect threw exception: System.InvalidOperationException: Negotiate failed with C00000BB.
   at Microsoft.Protocols.TestManager.SMBDPlugin.Detector.SMBDClient.Smb2Negotiate(DialectRevision[] requestDialects)
   at Microsoft.Protocols.TestManager.SMBDPlugin.Detector.SMBDDetector.TryNegotiateDialect(IPAddress ip, DialectRevision dialect)
[6/3/2020 1:10:10 PM] TryNegotiateDialect threw exception: System.InvalidOperationException: Negotiate failed with C00000BB.
   at Microsoft.Protocols.TestManager.SMBDPlugin.Detector.SMBDClient.Smb2Negotiate(DialectRevision[] requestDialects)
   at Microsoft.Protocols.TestManager.SMBDPlugin.Detector.SMBDDetector.TryNegotiateDialect(IPAddress ip, DialectRevision dialect)
[6/3/2020 1:10:10 PM] SMB dialects supported by SUT: Smb30
[6/3/2020 1:10:10 PM] Finished
[6/3/2020 1:10:10 PM] Get the remote adapters...
[6/3/2020 1:10:10 PM] Choose 10.1.1.79 as non-RDMA IP address.
[6/3/2020 1:10:10 PM] Choose 172.16.1.1 as RDMA IP address of SUT.
[6/3/2020 1:10:10 PM] Finished
[6/3/2020 1:10:10 PM] Connect to share through non-RDMA transport...
[6/3/2020 1:10:10 PM] Finished
[6/3/2020 1:10:10 PM] Connect to share through RDMA transport...
[6/3/2020 1:10:11 PM] Finished
[6/3/2020 1:10:11 PM] Check the supported SMBD capabilities of SUT...
[6/3/2020 1:10:11 PM] CheckSMBDNegotiate threw exception: System.InvalidOperationException: ConnectToServerOverRdma failed!
   at Microsoft.Protocols.TestManager.SMBDPlugin.Detector.SMBDClient.ConnectOverRDMA(String localIpAddress, String remoteIpAddress, Int32 port, UInt32 maxReceiveSize, RdmaAdapterInfo& adapterInfo)
   at Microsoft.Protocols.TestManager.SMBDPlugin.Detector.SMBDDetector.CheckSMBDNegotiate(RdmaAdapterInfo& rdmaAdapterInfo)
[6/3/2020 1:10:11 PM] Failed

**Comment by huiren-jiang** [comment]
Hi vinitagnihotri,

I’d like to double confirm if the previous FormatException from your test result log still exists or not after changing the values to 1 as I suggested, since only PTM auto-detection log is provided in your last reply. The auto-detection of PTM seems to fail for the same reason as your original issue. In order to investigate further, could you please send us the TRX file following the steps below?

1. Ensure the SMBD installer given in bullet 3 below is installed on your test driver. This installer added more logs which outputs more information when RDMA connect fails.

> Hi vinitagnihotri,
> 
> According to the log, some RDMA operations after opening adapter failed. This should be caused by mis-configuration of test suites, please try below steps:
> 
> 1. Some RDMA related configuration in `MS-SMBD_ServerTestSuite.ptfconfig` might be not applied after you installed the latest version. For more information, you could refer to the [RDMA Capability Settings](https://github.com/microsoft/WindowsProtocolTestSuites/blob/staging/TestSuites/MS-SMBD/docs/MS-SMBD_ServerUserGuide.md#configuring-the-test-suite) part of the MS-SMBD user guide.
> 2. Check the detailed RDMA log on your server to identify the reason of rejection, if available.
> 3. Uninstall the SMBD test suites, and install the [MS-SMBD-TestSuite-ServerEP.zip](https://github.com/microsoft/WindowsProtocolTestSuites/files/4507148/MS-SMBD-TestSuite-ServerEP.zip)
>    which we added more logs in commit [378ec1d](https://github.com/microsoft/WindowsProtocolTestSuites/commit/378ec1d7eeb97d79dacbf87e61361d02290dd29e). Set `RdmaLayerLoggingEnabled` to `true` and attach the new logs after re-run.
> 
> Best Regards,
> 
> Windows Protocol Test Suites Development Team

2. Download and extract [MS-SMBD_ServerTestSuite.ptfconfig.zip](https://github.com/microsoft/WindowsProtocolTestSuites/files/4755760/MS-SMBD_ServerTestSuite.ptfconfig.zip). Use `MS-SMBD_ServerTestSuite.ptfconfig` to replace the original one under `C:\MicrosoftProtocolTests\MS-SMBD\Server-Endpoint\3.20.1.0\Bin`. The values of `InboundEntries`, `OutboundEntries` and `InboundReadLimit` are all set to `1`, which works against our test environment of Mellanox. Also, we set `RdmaLayerLoggingEnabled` to `true` so that detailed RDMA layer log will be output.

3. Use `RunBVTTestCases.cmd` on desktop to run all BVT test cases and attach the output *.trx file.

Best Regards,

Windows Protocol Test Suites Development Team

**Comment by vinitagnihotri** [comment]
just an update, I installed MSFT-3.20.1.0 on one of our chelsio setup and it worked out of box with auto-detection. Attaching exported profile if it can give you any clue.
We still have no luck with mellanox setup with same params.
[smbd-msft3_20_1_0-profile.zip](https://github.com/microsoft/WindowsProtocolTestSuites/files/4764722/smbd-msft3_20_1_0-profile.zip)

**Comment by huiren-jiang** [comment]
Hi vinitagnihotri,

The `RunBVTTestCases.cmd` is on the desktop of the test driver. The PTM profile and etl trace you provided do not contain any diagnostic information for us.

We hope to know could you pass test cases in Chelsio environments? 

- If yes, there must be something wrong with Mellanox driver installation or configurations.

- If not, we need the detailed test log to know which system call fails in order to find more clues for you.

Based on your answer, we will provide you more guidance on how to continue.

Best Regards,

Windows Protocol Test Suites Development Team

**Comment by vinitagnihotri** [comment]
Hi,
Yes we can run test suite for chelsio without any issue ,its just we are facing issue running MSFT in our mellnox setup.

I could not run .cmd you mentioned as its producing errors. I am attaching log.
[cmd_log.txt](https://github.com/microsoft/WindowsProtocolTestSuites/files/4769151/cmd_log.txt)

**Comment by huiren-jiang** [comment]
Hi vinitagnihotri,

There is a bug in `RunBVTTestCases.cmd`. If you run by administrator, the launched vstest could not access system32 folder. We will fix it later, thanks for reporting it!

Since your Chelsio environment could pass all test cases, we could conclude that the issue should come from Mellanox driver installation or configuration. You could confirm whether RDMA works fine between Mellanox endpoints as below:

- Install the latest Mellanox driver for your Windows test driver. Also check the user manual to finish all necessary configurations.

- Download [NetworkDirect_DDK.zip](https://www.microsoft.com/en-us/download/details.aspx?id=26645) and extract. There are useful tools such as `ndping.exe` under `bin\`, which you could use to check the connection since they use the same interface as our SMBD test suites.

Best Regards,

Windows Protocol Test Suites Development Team

**Comment by vinitagnihotri** [comment]
Ok i can try with zip file. However please note we can access our server over SMBDirect from windows explorer without any issue over MLX setup, therefore we are wondering why SMBD test suite faces issue, when microsoft explorer can access same share without any issue.

Thanks

**Comment by vinitagnihotri** [comment]
Here is result from ndrping.exe

Server:
[root@pg000pyx vinit1]# rping -s -p 11210
Received bogus data, size 0
recv wc error: -1
wait for RDMA_READ_ADV state 11
[root@pg000pyx vinit1]#

Windows 2012R2 client

E:\NetDirect\bin\amd64>ndrping.exe c 172.16.1.1:11210
Result 000000000173FE60 (Results array at 0000000001700080), failed with c0000120

**Comment by huiren-jiang** [comment]
Hi vinitagnihotri,

First we hope to double confirm,  does the old test suite (3.18.6.0) work well on the same computer? If so, since we did not change code related to RDMA SDK, did you change any configuration on the Mellanox driver when you were using the new test suite?

> Ok i can try with zip file. However please note we can access our server over SMBDirect from windows explorer without any issue over MLX setup, therefore we are wondering why SMBD test suite faces issue, when microsoft explorer can access same share without any issue.
> 
> Thanks

The SMB2/SMBD implementation of Windows uses kernel-mode interface provided by system directly, which finally calls into device driver of Mellanox. Our SMBD test suites uses user-mode Network Direct interface, which finally calls into mlx4nd.dll. So if there is something wrong with user-mode binaries or configurations, you might find that you could not run test cases through Mellanox, even if you could access the share from explorer.

Also, please note even if Mellanox driver is not configured properly, you might still access the share through it. That’s because the adapter could still be used as Ethernet adapter, so you need to confirm all the SMB2 traffic is transferred by RDMA instead of TCP.

> Here is result from ndrping.exe
> 
> Server:
> [root@pg000pyx vinit1]# rping -s -p 11210
> Received bogus data, size 0
> recv wc error: -1
> wait for RDMA_READ_ADV state 11
> [root@pg000pyx vinit1]#
> 
> Windows 2012R2 client
> 
> E:\NetDirect\bin\amd64>ndrping.exe c 172.16.1.1:11210
> Result 000000000173FE60 (Results array at 0000000001700080), failed with c0000120

It looks like the user-mode driver on your client side does not work properly. Since our test suites use the same system interface, ping/pong should fail similarly as our test code tries to connect over RDMA. You could try re-install the latest Mellanox driver for your adapter and OS. If the error continues, we suggest you to find Mellanox support for help.

Best Regards,

Windows Protocol Test Suites Development Team

**Comment by vinitagnihotri** [comment]
1. 3.18.6.0 worked on other setup, we didn't had mlx setup by then
2.We confirm traffic goes via SMDDirect as we use windows performance monitor to watch RDMA activity
3.We are using connectx-2 h/w (EOS now), but have latest drivers/firmware.

**Comment by huiren-jiang** [comment]
Hi vinitagnihotri,

Actually our Mellanox adapter in use is also ConnectX-2 IPoIB, you could refer to [Tested RDMA Adapter](https://github.com/microsoft/WindowsProtocolTestSuites/blob/staging/TestSuites/MS-SMBD/docs/MS-SMBD_ServerUserGuide.md#tested-rdma-adapter). For recent versions of SMBD test suites, we run regression tests on Windows Server 2016 which you could get from [Evaluation Center](https://www.microsoft.com/en-us/evalcenter/evaluate-windows-server-2016). You could try to install it with the driver given in our user guide, everything should work as expected and no more configuration is needed.

Also, we will try on Windows Server 2012R2 since you are currently using it. We will let you know once the result is ready.

Best Regards,

Windows Protocol Test Suites Development Team

**Comment by vinitagnihotri** [comment]
Hi,
We installed same driver now on our setup but it results into same behavior. Unfortunately I won't be able to use version 2016 for this testing as of now, so we have to stay put with 2012R2 for a while.

**Comment by huiren-jiang** [comment]
Hi vinitagnihotri,

As I tested against a new clean install of Windows Server 2012R2, all basic test cases could pass as expected for Mellanox adapter.

So far, we have verified that the tests could pass in our environment with the same version of adapter, OS, driver and test suites as yours. Please try on other machine or have a new clean install when you have time.

Best Regards,

Windows Protocol Test Suites Development Team

**Comment by vinitagnihotri** [comment]
Alright. I believe we did all we can.
I shall try to install different setup and try.
I am closing this issue as of now.
Thank you for every help.
